f2py generated C code doesn't recognize "long_long"

When using a .f2py_f2cmap to specify 64 bit integers (int64 from iso_fortran_env) as "long_long", the code will not compile if any of the subroutine parameters are specified as int64 (as long_long appears in the generated c-code for the function).  

If using a local variable with the int64 kind the error does not occur.

Generating the C code, modifying "long_long" to "long long" does allow compilation.

### Reproducing code example:

test.f95
```fortran
      subroutine func1(n, x, res)
        use, intrinsic :: iso_fortran_env, only: int64, real64
        implicit none
        integer(int64), intent(in) :: n
        real(real64), intent(in) :: x(n)
        real(real64), intent(out) :: res
Cf2py   intent(hide) :: n

        res = sum(x)
      end
```

.f2py_f2cmap
```
dict(real=dict(real32='float', real64='double'), integer=dict(int64='long_long'))
```

Compiling with GCC (11.0.0)/GFortran(9.2.0) on MacOS 10.15.1:
```shell script
f2py -c -m test test.95
```

<!-- Remove these sections for a feature request -->

### Error message:
```shell script
.../testmodule.c:141:33: error: unknown type name 'long_long'
extern void F_FUNC(func1,FUNC1)(long_long*,double*,double*);
                                ^
.../testmodule.c:163:46: error: unknown type name 'long_long'
                           void (*f2py_func)(long_long*,double*,double*)) {
                                             ^
.../testmodule.c:168:3: error: use of undeclared identifier 'long_long'
  long_long n = 0;
```
<!-- Full error message, if any (starting from line Traceback: ...) -->

### Numpy/Python version information:

- Numpy: 1.17.4
- Python: 3.7.5

--- a/numpy/f2py/capi_maps.py
+++ b/numpy/f2py/capi_maps.py
@@ -176,6 +176,7 @@ f2cmap_all = {'real': {'': 'float', '4': 'float', '8': 'double',
 
 f2cmap_default = copy.deepcopy(f2cmap_all)
 
+f2cmap_mapped = []
 
 def load_f2cmap_file(f2cmap_file):
     global f2cmap_all
@@ -212,6 +213,7 @@ def load_f2cmap_file(f2cmap_file):
                     f2cmap_all[k][k1] = d[k][k1]
                     outmess('\tMapping "%s(kind=%s)" to "%s"\n' %
                             (k, k1, d[k][k1]))
+                    f2cmap_mapped.append(d[k][k1])
                 else:
                     errmess("\tIgnoring map {'%s':{'%s':'%s'}}: '%s' must be in %s\n" % (
                         k, k1, d[k][k1], d[k][k1], list(c2py_map.keys())))

--- a/numpy/f2py/rules.py
+++ b/numpy/f2py/rules.py
@@ -1323,6 +1323,9 @@ def buildmodule(m, um):
         rd = dictappend(rd, ar)
 
     needs = cfuncs.get_needs()
+    # Add mapped definitions
+    needs['typedefs'] += [cvar for cvar in capi_maps.f2cmap_mapped #
+                          if cvar in typedef_need_dict.values()]
     code = {}
     for n in needs.keys():
         code[n] = []

--- a/numpy/f2py/tests/src/f2cmap/.f2py_f2cmap
+++ b/numpy/f2py/tests/src/f2cmap/.f2py_f2cmap
@@ -1,0 +1,1 @@
+dict(real=dict(real32='float', real64='double'), integer=dict(int64='long_long'))

--- a/numpy/f2py/tests/src/f2cmap/isoFortranEnvMap.f90
+++ b/numpy/f2py/tests/src/f2cmap/isoFortranEnvMap.f90
@@ -1,0 +1,9 @@
+      subroutine func1(n, x, res)
+        use, intrinsic :: iso_fortran_env, only: int64, real64
+        implicit none
+        integer(int64), intent(in) :: n
+        real(real64), intent(in) :: x(n)
+        real(real64), intent(out) :: res
+Cf2py   intent(hide) :: n
+        res = sum(x)
+      end

--- a/numpy/f2py/tests/test_f2cmap.py
+++ b/numpy/f2py/tests/test_f2cmap.py
@@ -1,0 +1,15 @@
+from . import util
+import numpy as np
+
+class TestF2Cmap(util.F2PyTest):
+    sources = [
+        util.getpath("tests", "src", "f2cmap", "isoFortranEnvMap.f90"),
+        util.getpath("tests", "src", "f2cmap", ".f2py_f2cmap")
+    ]
+
+    # gh-15095
+    def test_long_long_map(self):
+        inp = np.ones(3)
+        out = self.module.func1(inp)
+        exp_out = 3
+        assert out == exp_out

